libevm/VMCalls.cpp

```diff
@@ -182,14 +189,17 @@ bool VM::caseCallSetup(CallParameters *callParams, bytesRef& o_output)
 {
 	m_runGas = toInt63(m_schedule->callGas);
 
-	if (m_OP == Instruction::CALL && !m_ext->exists(asAddress(m_SP[1])))
+	callParams->staticCall = (m_OP == Instruction::STATICCALL || m_ext->staticCall);
+
+	Address destinationAddr = asAddress(m_SP[1]);
+	if (m_OP == Instruction::CALL && !m_ext->exists(destinationAddr))
 		if (m_SP[2] > 0 || m_schedule->zeroValueTransferChargesNewAccountGas())
 			m_runGas += toInt63(m_schedule->callNewAccountGas);
 
-	if (m_OP != Instruction::DELEGATECALL && m_SP[2] > 0)
+	if ((m_OP == Instruction::CALL || m_OP == Instruction::CALLCODE) && m_SP[2] > 0)
```

Comment:
I hope this won't get us into trouble: We change for value transfer for callcode, although we do not consider it state changing.